src/pages/profile/password.ts

```diff
@@ -0,0 +1,110 @@
+import { Block, PathRouter, Store } from 'core';
+import { ValidateRuleType } from 'helpers/validateForm';
+import ControlledInput from 'components/controlledInput';
+import Layout from 'components/layout';
+import { withRouter } from 'helpers/withRouter';
+import { withStore } from 'helpers/withStore';
+import { Params } from 'core/router/PathRouter';
+import { editPassword, login } from "services/auth";
+
+type LoginProps = {
+  redirectToProfile?: (e: MouseEvent) => void;
+  onSubmit?: (e: FormDataEvent) => void;
+  onInput?: () => void;
+  onFocus?: () => void;
+  onBlur?: () => void;
+  setErrorValidation?: (val: boolean) => void;
+  router: PathRouter;
+  store: Store<AppState>;
+  params: Params;
+  formError?: () => string | null;
+  formValues: { oldPassword: string; newPassword: string };
+};
+
+export class EditPasswordPage extends Block<LoginProps> {
+  static componentName = 'EditPasswordPage';
+  constructor(props?: LoginProps) {
+    super(props);
+    this.state = { validationError: false };
+    const defFormValues = { oldPassword: '', newPassword: '' };
+    this.setProps({
+      ...this.props,
+      redirectToProfile: (e: MouseEvent) => this.redirectToProfile(e),
+      onSubmit: (e: FormDataEvent) => this.onSubmit(e),
+      setErrorValidation: (val: boolean) => {
+        this.setState({ validationError: val });
+      },
+      formValues: defFormValues,
+      formError: () => this.props.store.getState().loginFormError,
+    });
+  }
+
+  redirectToProfile = (e: MouseEvent) => {
+    e.preventDefault();
+    this.props.router.go('/setting');
+  };
+
+  onSubmit(e: FormDataEvent) {
+    e.preventDefault();
+    this.setState({ validationError: false });
+    const event = new Event('blur');
+    const inputList = this._element?.querySelectorAll(
+      'input'
+    ) as NodeListOf<HTMLInputElement>;
+    inputList.forEach((item: HTMLInputElement) => {
+      item.dispatchEvent(event);
+    });
+    const formData: any = {};
+    if (!this.state.validationError) {
+      inputList.forEach((item: HTMLInputElement) => {
+        formData[`${item.name}`] = item.value;
```

Comment:
Использование шаблонной строки здесь не нужно. Так тоже будет работать:
```suggestion
        formData[item.name] = item.value;
```